bridges/prometheus/producer_test.go

```diff
@@ -251,3 +252,85 @@ func TestProduce(t *testing.T) {
 		})
 	}
 }
+
+func TestProduceForStartTime(t *testing.T) {
+	testCases := []struct {
+		name        string
+		testFn      func(*prometheus.Registry)
+		startTimeFn func(metricdata.Aggregation) []time.Time
+	}{
+		{
+			name: "counter",
+			testFn: func(reg *prometheus.Registry) {
+				metric := prometheus.NewCounter(prometheus.CounterOpts{
+					Name: "test_counter_metric",
+					Help: "A counter metric for testing",
+					ConstLabels: prometheus.Labels(map[string]string{
+						"foo": "bar",
+					}),
+				})
+				reg.MustRegister(metric)
+				metric.(prometheus.ExemplarAdder).AddWithExemplar(
+					245.3, prometheus.Labels{
+						"trace_id":        traceIDStr,
+						"span_id":         spanIDStr,
+						"other_attribute": "abcd",
+					},
+				)
+			},
+			startTimeFn: func(aggr metricdata.Aggregation) []time.Time {
+				dps := aggr.(metricdata.Sum[float64]).DataPoints
+				sts := make([]time.Time, len(dps))
+				for i, dp := range dps {
+					sts[i] = dp.StartTime
+				}
+				return sts
+			},
+		},
+		{
+			name: "histogram",
+			testFn: func(reg *prometheus.Registry) {
+				metric := prometheus.NewHistogram(prometheus.HistogramOpts{
+					Name: "test_histogram_metric",
+					Help: "A histogram metric for testing",
+					ConstLabels: prometheus.Labels(map[string]string{
+						"foo": "bar",
+					}),
+				})
+				reg.MustRegister(metric)
+				metric.(prometheus.ExemplarObserver).ObserveWithExemplar(
+					578.3, prometheus.Labels{
+						"trace_id":        traceIDStr,
+						"span_id":         spanIDStr,
+						"other_attribute": "efgh",
+					},
+				)
+			},
+			startTimeFn: func(aggr metricdata.Aggregation) []time.Time {
+				dps := aggr.(metricdata.Histogram[float64]).DataPoints
+				sts := make([]time.Time, len(dps))
+				for i, dp := range dps {
+					sts[i] = dp.StartTime
+				}
+				return sts
+			},
+		},
+	}
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			reg := prometheus.NewRegistry()
+			tt.testFn(reg)
+			p := NewMetricProducer(WithGatherer(reg))
+			output, err := p.Produce(context.Background())
+			assert.NoError(t, err)
+			for _, sms := range output {
```

Comment:
This test would pass if output or their metrics aren't empty, yet there wouldn't be anything tested.

How about an assert.NotEmpty before each loop?

Reply:
I fixed it.
https://github.com/open-telemetry/opentelemetry-go-contrib/compare/849e455b517da4332ebe1a39506ddc858e422e2b..b125e660546eb80511082f56ef4f4cb400ebb56f